scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -542,6 +542,24 @@ def get_platform(self, name):
                 break
         return selected_platform
 
+    def handle_quarantined_tests(self, instance: TestInstance, plat: Platform):
+        matched_quarantine = False
+        if self.quarantine:
+            matched_quarantine = self.quarantine.get_matched_quarantine(
+                instance.testsuite.id, plat.name, plat.arch, plat.simulation
+            )
+            if matched_quarantine and not self.options.quarantine_verify:
+                instance.add_filter("Quarantine: " + matched_quarantine, Filters.QUARANTINE)
+                return
+            if not matched_quarantine and self.options.quarantine_verify:
+                instance.add_filter("Not under quarantine", Filters.QUARANTINE)
+                return
+        if not matched_quarantine and instance.reason:
```

Comment:
No! When you want to test scenarios from the quarantine, then you can use:
`--quarantine-list quarantine.yaml --quarantine-verify`

With change you proposed, you ask user to use the quarantine file every time with `--test-only`.
When someone run tests with quarantine file, and then want to rerun it with '--test-only` (or `--only-failed`) and does not provide same options for quarantine, then twister fails, because there is no artefacts in `twister-out` directory.

Just try with commands:
scripts/twister -vv -p native_sim -T samples/hello_world  --quarantine-list quarantine.yaml
scripts/twister -vv -p native_sim -T samples/hello_world --test-only

Reply:
I remove the change and add comments on how to reenable cases in commit message. Thanks again.